packages/foam-vscode/src/dated-notes.ts

```diff
@@ -1,14 +1,15 @@
 import { workspace, WorkspaceConfiguration } from 'vscode';
-import { dirname, join } from 'path';
+import { dirname, isAbsolute } from 'path';
 import dateFormat from 'dateformat';
 import * as fs from 'fs';
 import { docConfig, focusNote, pathExists } from './utils';
+import { URI } from 'foam-core';
 
 async function openDailyNoteFor(date?: Date) {
   const foamConfiguration = workspace.getConfiguration('foam');
   const currentDate = date !== undefined ? date : new Date();
 
-  const dailyNotePath = getDailyNotePath(foamConfiguration, currentDate);
+  const dailyNotePath = getDailyNotePath(foamConfiguration, currentDate).fsPath;
```

Comment:
one more comment about `URI`, as we want to use it wherever possible, until the last moment when we actually need the `fsPath`

Basically `createDailyNoteIfNotExists` should receive a `URI`, and when calling `writeFile` translate it into the corresponding `fsPath`

the goal is to never see an `fsPath` in the codebase which is not a direct argument to a `fs` call

Reply:
I've added a new commit that changes the signatures of all these methods to use URI instead of string for the paths.
Then the changes related to allowing absolute directories is done in a separate commit on top of that.